### PR TITLE
feat: add `useStrapi` composable with correct types

### PR DESCRIPTION
**Pull request by harlan-zw**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This PR aims to fix broken types without introducing breaking changes.

See: 
- https://github.com/nuxt-modules/strapi/issues/268
- https://github.com/nuxt-modules/strapi/issues/274

To summarise the type issues:
- single/multiple return requests have ambiguous types
- it's not possible to type the update / create functions without an error

To fix the types without breaking anything, I've introduced a new composable `useStrapi` which is a simple wrapper to the existing v3 or v4 client, depending on which version is being used, with the correct types.

The types for the client allow a user to either set a default type for all requests from the composable `useStrapi<MyData>` and / or override the single data return types from the functions `find<Book>('book')`

`useStrapi3` and `useStrapi4` are deprecated in favour of this to notify users they should swap over, this can be removed but I think it's worth while to have a single composable in the docs which have correct types

I believe these changes are a huge DX improvement:
- No type importing
- Single and multiple return types are automatically handled 
- Proper types for CRUD
- Single composable, makes upgrading easier

While this shouldn't break anything, it should be a minor bump and simple migration docs should be included in the release

## :warning: Note

While these changes have minimal scope for breaking things there are no test cases and I've never used strapi, so ideally someone would test this PR locally before merging.

## Migration

Check the [docs](https://strapi.nuxtjs.org/usage) for latest type guidelines.

**Previous version**

```ts
import type { Strapi4Response } from '@nuxtjs/strapi'
const { find } = useStrapi4()
const response = await find<Strapi4Response<Restaurant>>('restaurants')
```

**New version**

```ts
const { find } = useStrapi()
const response = await find<Restaurant>('restaurants')
```

or

```ts
const { find } = useStrapi<Restaurant>()
const response = await find('restaurants')
```

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes (if not applicable, please state why) - **no test suit**

### CODE REVIEW

File: src/runtime/composables-v3/useStrapi.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  count(contentType: string, params?: Strapi3RequestParams): Promise<number>
+  find<F = T[]>(contentType: string, params?: Strapi3RequestParams): Promise<F>
+  findOne<F = T>(contentType: string, id: string | number, params?: Strapi3RequestParams): Promise<F>
+  create<F = T>(contentType: string, data: Partial<F>): Promise<F>
+  update<F = T>(contentType: string, data: Partial<F>): Promise<F>
+  delete<F = T>(contentType: string, id: string | number | Partial<F>, data?: Partial<F>): Promise<F>
+}
+
+/**
+ * @deprecated use `useStrapi` for correct types
```
Comment (benjamincanac) [inline src/runtime/composables-v3/useStrapi.ts:14]: Why is there a deprecated here?

**Comment by benjamincanac** [comment]
Hello @harlan-zw, thanks for your PR it's awesome! I've made two commits to fix the lint and removed a deprecated, can you let me know what you think? Otherwise the PR is good for me! :)

File: src/runtime/composables-v3/useStrapi.ts (hunk 1 above)
Comment (harlan-zw) [reply src/runtime/composables-v3/useStrapi.ts]: `useStrapi` calls this function, just with the right types. So using it directly, with typescript, is going to be buggy for people

**Comment by harlan-zw** [comment]
Thanks for the review @benjamincanac!

Commits look good, I'm not too fussed about the removal of the `@deprecated` flag, it just acts as a way to push people towards using the function with the correct types, without introducing breaking changes. If it wasn't obvious, useStrapi is the same as useStrapi{version})

Did you test this locally with an actual strapi API by any chance? I would have liked to add tests for it but it was missing a suite and was too much work to add it.

**Comment by benjamincanac** [comment]
I removed the `deprecated` on `useStrapi` composable not `useStrapi3`, I'll test this on my projects once we merge it through the edge channel!

File: src/runtime/composables-v3/useStrapi.ts (hunk 1 above)
Comment (benjamincanac) [reply src/runtime/composables-v3/useStrapi.ts]: Sorry, I might be misunderstanding but isn't this `useStrapi`?

Comment (Atinux) [reply src/runtime/composables-v3/useStrapi.ts]: Indeed I think it’s not needed here

**Pull request merged**